src/components/common/StaffPayment.vue

```diff
@@ -4,17 +4,88 @@
       <h2>3. Staff Payment</h2>
     </header>
 
-    <div id="staff-payment-container">
-      Under construction!
-    </div>
+    <staff-payment-component
+      :staffPaymentData="getStaffPayment"
+      @update:staffPaymentData="onStaffPaymentData"
+      @valid="staffPaymentFormValid=$event"/>
   </section>
 </template>
 
 <script lang="ts">
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Emit, Vue, Watch } from 'vue-property-decorator'
+import { Action, Getter } from 'vuex-class'
 
-@Component({})
+// Components
+import { StaffPayment as StaffPaymentComponent } from '@bcrs-shared-components/staff-payment'
+
+// Mixins, Interfaces and Enums
+import { FilingTemplateMixin } from '@/mixins'
+import { ActionBindingIF } from '@/interfaces'
+
+import { StaffPaymentIF } from '@bcrs-shared-components/interfaces'
+import { StaffPaymentOptions } from '@bcrs-shared-components/enums'
+
+@Component({
+  components: {
+    StaffPaymentComponent
+  }
+})
 export default class StaffPayment extends Vue {
+  // Global getters
+  @Getter getStaffPayment!: StaffPaymentIF
+
+  // Global setters
+  @Action setStaffPayment!: ActionBindingIF
+
+  private staffPaymentFormValid: boolean = null
+
+  /** Emits Valid event. */
+  @Emit('emitValid')
+  private emitValid (): boolean {
+    return this.staffPaymentFormValid
+  }
+
+  private onStaffPaymentData (event) {
+    let staffPaymentData: StaffPaymentIF = { ...this.getStaffPayment, ...event }
+    switch (staffPaymentData.option) {
+      case StaffPaymentOptions.FAS:
+        staffPaymentData = {
+          option: StaffPaymentOptions.FAS,
+          routingSlipNumber: staffPaymentData.routingSlipNumber,
+          isPriority: staffPaymentData.isPriority,
+          bcolAccountNumber: '',
```

Comment:
In Filings UI we don't include these properties and I'm not sure what the API will do with them present (even if empty).
I suggest you update the IF to specify which properties are optional, and then don't specify them here (ditto in BCOL and NO_FEE cases).

Reply:
It won't be included in API call. That has been handled in filing-template-mixin.ts

Reply:
Hmm, you shouldn't be getting a TS error since those properties are defined as optional.

Reply:
True, but it was not optional in shared component. That's how it is been implemented in filing-ui.

Reply:
Ah, OK.
It would be good for the IF to be the same in the component and in this project, yes?

Reply:
OK for now. We'll resolve this later.